stubs/assertpy/assertpy/exception.pyi

```diff
@@ -5,4 +5,4 @@ __tracebackhide__: bool
 
 class ExceptionMixin:
     def raises(self, ex: type[BaseException] | BaseException) -> Self: ...
-    def when_called_with(self, *some_args: Any, **some_kwargs: dict[str, Any]) -> Self: ...
+    def when_called_with(self, *some_args: Any, **some_kwargs: Any) -> Self: ...
```

Comment:
Thanks! Since it's now our policy to document each `Any`, maybe we could use this opportunity to add a comment. I'm not familiar with assertpy, but maybe something like this?

```suggestion
    # The types of some_args and some_kwargs must equal the types of the called function.
    def when_called_with(self, *some_args: Any, **some_kwargs: Any) -> Self: ...
```

(This might even be case where we could make `ExceptionMixin` generic over `val` in the future.)